src/ProductConstructionService/ProductConstructionService.Api/Configuration/PcsConfiguration.cs

```diff
@@ -53,6 +54,11 @@ public static string GetRequiredValue(this IConfiguration configuration, string
 
         string databaseConnectionString = builder.Configuration.GetRequiredValue(DatabaseConnectionString);
 
+        if (appendUserId)
```

Comment:
It's unclear where `appendUserId` would be appended from the scope of the whole method (when you're calling it and that it's connected to SQL).

How about instead, we do `.Replace()` for a placeholder in the connection string and either the placeholder is present or not based on the appsettings value?

Reply:
I like that, it's elegant, done